google/cloud/bigquery/dbapi/_helpers.py

```diff
@@ -20,15 +20,34 @@
 import numbers
 
 from google.cloud import bigquery
-from google.cloud.bigquery import table
+from google.cloud.bigquery import table, enums
 from google.cloud.bigquery.dbapi import exceptions
 
 
 _NUMERIC_SERVER_MIN = decimal.Decimal("-9.9999999999999999999999999999999999999E+28")
 _NUMERIC_SERVER_MAX = decimal.Decimal("9.9999999999999999999999999999999999999E+28")
+_BIGQUERY_SCALAR_TYPES = enums.SqlParameterScalarTypes.__dict__
 
 
-def scalar_to_query_parameter(value, name=None):
+def _parameter_type(name, value, query_parameter_type=None, value_doc=""):
+    if query_parameter_type:
+        parameter_type = query_parameter_type.upper()
+        if parameter_type not in _BIGQUERY_SCALAR_TYPES:
+            raise exceptions.ProgrammingError(
+                f"The given parameter type, {query_parameter_type},"
+                f" for {name} is not a valid BigQuery scalar type."
+            )
```

Comment:
We might want to do something like this to make sure our aliases work as expected:

```suggestion
            )
            parameter_type = getattr(enums.SqlParameterScalarTypes, parameter_type)._type
```

Reply:
Oops, yup.

Reply:
Pushed.